docs/graphql/manual/auth/authorization/role-multiple-rules.rst

```diff
@@ -0,0 +1,119 @@
+
+We want the role ``user`` to be able to access:
+
+- the ``email``, ``phone`` and ``address`` columns only if the ``id`` column is the requesting user's id, i.e. current
+  user is owner of the row.
+
+- the ``id``, ``name`` and ``city`` columns for all rows.
+
+We can achieve this via the following steps:
+
+Step 1: Create a view 
+---------------------
+
+:ref:`Create <create_views>` a view ``user_private`` with columns ``(user_id, email, phone, address)``:
+
+.. code-block:: SQL
+
+    CREATE VIEW user_private AS
+      SELECT id AS user_id, email, phone, address
+        FROM user_info;
+
+Step 2: Create a relationship
+-----------------------------
+
+For the table ``user_info``, :ref:`create <create_manual_relationships>` a manual object relationship called
+``private_info`` using ``user_info : id -> user_private : user_id``:
+
+.. thumbnail:: ../../../../img/graphql/manual/auth/multiple-rules-create-manual-relationship.png
+
+Step 3: Define permissions
+--------------------------
+
+For the role ``user``, create the following permissions for ``select``:
+
+- Table ``user_info``: allow access to ``id``, ``name`` and ``city`` without any row conditions.
+
+.. thumbnail:: ../../../../img/graphql/manual/auth/multiple-rules-define-public-permissions.png
+
+- View ``user_private``: allow access to ``id``, ``phone``, ``email`` and ``address`` if the ``user-id``
+  passed in the session variable is equal to row's ``user_id``
+
+.. thumbnail:: ../../../../img/graphql/manual/auth/multiple-rules-define-private-permissions.png
+
+Step 4: Query with appropriate access control
+---------------------------------------------
+
+Now we can fetch the required data with the appropriate access control by using the relationship.
+
+If the ``X-Hasura-Role`` and the ``X-Hasura-User-Id`` session variables are set to ``user`` and ``2`` respectively:
+
+.. graphiql::
+  :view_only:
+  :query:
+    query {
+      user_info {
+        id
+        name
+        city
+        private_info {
+          email
+          phone
+          address
+        }
+      }
+    }
+  :response:
+    {
+      "data": {
+        "user_info": [
+          {
+            "id": 1,
+            "name": "Julie",
+            "city": "Boston",
+            "private_info": null
+          },
+          {
+              "id": 2,
+              "name": "Josh",
+              "city": "Bangalore",
+              "private_info": {
+                "email": "josh@josh.com",
+                "phone": "+91-9787675678",
+                "address": "#141, 7th Main Road, Koramangala 3rd Block",
+              }
+            },
+            {
+              "id": 3,
+              "name": "John",
+              "city": "Berlin",
+              "private_info": null
+            }
+        ]
+      }
+    }
+
+Note that the ``private_info`` field is returned as ``null`` for all rows without the appropriate access.
```

Comment:
Should we make this as a note? I think this is generally useful go highlight.

Reply:
Yeah true. But notes and admonition we generally use only to highlight generic information. Putting the result of an example as a note feels a little odd. Maybe some sort of bolding might help but not sure how exactly. Let me know if you have an idea or else we can leave it as is I guess.

Reply:
That makes sense. I guess what we could do is make it a generic note informing about the behaviour. But we can also leave it 🙂